test/jit/test_pdt.py

```diff
@@ -0,0 +1,189 @@
+import os
+import sys
+import torch
+from torch.testing._internal.jit_utils import JitTestCase, make_global
+from torch.jit._monkeytype_config import _IS_MONKEYTYPE_INSTALLED
+from typing import List, Dict, Tuple  # noqa: F401
+
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+
+if not _IS_MONKEYTYPE_INSTALLED:
+    print("monkeytype is not installed. Skipping tests for Profile-Directed Typing", file=sys.stderr)
+    JitTestCase = object  # type: ignore[misc, assignment] # noqa: F811
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+class TestPDT(JitTestCase):
+    """
+    A suite of tests for profile directed typing in TorchScript.
+    """
+    def test_pdt(self):
+        def test_sum(a, b):
+            return a + b
+
+        make_global(test_sum)
+        scripted_fn_add = torch.jit._script_pdt(test_sum, example_inputs=[(3, 4)])
+        self.assertEqual(scripted_fn_add(10, 2), test_sum(10, 2))
+
+        def test_sub(a, b):
+            return a - b
+
+        make_global(test_sub)
+        scripted_fn_sub = torch.jit._script_pdt(test_sub, example_inputs=[(3.9, 4.10)])
+        self.assertEqual(scripted_fn_sub(6.5, 2.9), test_sub(6.5, 2.9))
+
+        def test_mul(a, b):
+            return a * b
+
+        make_global(test_mul)
+        scripted_fn_mul = torch.jit._script_pdt(test_mul, example_inputs=[(-10, 9)])
+        self.assertEqual(scripted_fn_mul(-1, 3), test_mul(-1, 3))
+
+        def test_args_complex(real, img):
+            return torch.complex(real, img)
+
+        make_global(test_args_complex)
+        scripted_fn_complex = torch.jit._script_pdt(test_args_complex, example_inputs=[(torch.rand(3, 4), torch.rand(3, 4))])
+        arg1, arg2 = torch.rand(3, 4), torch.rand(3, 4)
+        self.assertEqual(scripted_fn_complex(arg1, arg2), test_args_complex(arg1, arg2))
+
+        def test_bool(a):
+            if a:
+                return -1
+            else:
+                return 0
+
+        make_global(test_bool)
+        scripted_fn_bool = torch.jit._script_pdt(test_bool, example_inputs=[(True,)])
+        self.assertEqual(scripted_fn_bool(True), test_bool(True))
+
+        def test_str(a):
+            if a == "":
+                return False
+            else:
+                return True
+
+        make_global(test_str)
+        scripted_fn_str = torch.jit._script_pdt(test_str, example_inputs=[("",)])
+        self.assertEqual(scripted_fn_str("abc"), test_str("abc"))
+
+    def test_pdt_list_and_tuple(self):
```

Comment:
Could you try some nested data structure as well? like List[Tuple] or List[List]